### DISCUSSION TITLE
MacOS specific renderer crashing on specific websites

### DISCUSSION THREAD

**Discussion by slowengine** [ultrafunkamsterdam/undetected-chromedriver#1080] [Q&A]
A few weeks ago I stumbled upon a page that would crash UC when running on a MacOS system.
While I have not tested it on Linux, windows does not seem to exhibit the problem.

The crashing issue is reproducible at least a few chrome versions back.
It is also not the problem of the patched binary, nor the difference between mac64 and mac_arm64 architectures.

But the same renderer crashing does not happen with vanilla selenium. Which means that it is specific to whatever UC does when interacting with chromedriver and only on MacOS.

OS: MacOS Monterey
Python: 3.8, 3.9(pre-installed), 3.11
UC: 3.4.6
Chrome: 108-110

To reproduce the issue on mac, run the following code:
```python
import undetected_chromedriver as uc
options = uc.ChromeOptions()
options.add_experimental_option('prefs', {'intl.accept_languages': 'en'})
driver = uc.Chrome(options=options)
driver.get('https://www.patreon.com/posts/updated-in-2022-66910536')
```

I did test it with native python installation and also brew versions of python, but all of them show the same behaviour. Same DOM mangling, same renderer crash. The issue relating to this is #1049, which had a bad timing.

If you know more pages that cause that type of crashing, do post them, maybe something can be correlated.
But for now I can't figure out what is causing this issue, so if you have any ideas, I'm all ears.

Thanks.

**Comment by dogi-sol** [comment]
I had a similar error, it was only solved by downgrading to version 3.1.2.
Site: https://magiceden.io/

**Reply by slowengine** [reply]
Do you still remember from what version you downgraded? Since there are quite a lot of changes since 3.1.2 and HEAD. Maybe it will give an indication what specific part is causing this. Although as I continue to document what I find in #1049, it seems that disconnecting from the chromedriver allows the the page to finish loading.

**Reply by dogi-sol** [reply]
Unfortunately, I don't remember the exact version. However, I do remember that everything was working for a long time and then suddenly stopped. Perhaps the Chrome version 110 came out around that time. But it's also possible that something was changed on the website itself that caused this to happen. I'm using a M2 MacBook Air with OS version 12.6.

**Reply by dogi-sol** [reply]
Unfortunately, since Datadome was added to the website, I haven't been able to use it because it's being detected. I also tried the feature branch fix found in the open pull requests, but it didn't help with this problem. (https://github.com/ultrafunkamsterdam/undetected-chromedriver/pull/1010)
